### DISCUSSION TITLE
Wiki: digital drawing of the gameplay clock stack (made with LibreOffice)

### DISCUSSION THREAD

**Discussion by Speykious** [ppy/osu#14643] [Ideas]
I was reading the wiki to understand how osu!lazer works ─ I thought it would be great to have a digital drawing of the gameplay clock stack for https://github.com/ppy/osu/wiki/Gameplay-clocks.
How is that?

![image](https://user-images.githubusercontent.com/13885008/132143072-9a1e2d2a-a7c4-4ffe-b75f-7843a63abb91.png)

**Comment by smoogipoo** [comment]
Just note that the structure has changed in recent times - there's a new `GameplayClockContainer` which houses everything and caches the `GameplayClock`, and the bottom structure doesn't exist in `RulesetInputManager` anymore (`FrameStabilityContainer`). There's also a lot more interaction around spectating that is undocumented (primarily, `WaitingForFrames`).

**Reply by Speykious** [reply]
OK. I can change the drawing at any point, I have it saved. I will give the current version of the `.odg` later today so that it's easier to deal with if anything.
